File: Python/String-2/start_out.py

# return a version of the given string, where for every star (*) in the string the star and the chars immediately to its left and right are gone
# so "ab*cd" yields "ad" and "ab**cd" also yields "ad"
def start_out(str):
    s = ''

    for x in range(len(str)):
        if x == 0 and str[x] != '*':
            s = s + str[x]
        if x > 0 and str[x] != '*' and str[x-1] != '*':
            s = s + str[x]
        if x > 0 and str[x] == '*' and str[x-1] != '*':
            s = s[:len(s)-1]
    
    return s

print(start_out('ab*cd'))
print(start_out('ab**cd'))
print(start_out('sm*eilly'))

File: JavaScript/String-2/startOut.js

// return a version of the given string, where for every star (*) in the string the star and the chars immediately to its left and right are gone
// so 'ab*cd' yields 'ad' and 'ab**cd' also yields 'ad'
const startOut = (str) => {
	let newStr = '';

	for (let i = 0; i < str.length; i++) {
		if (str[i] === '*' || i > 0 && str[i - 1] === '*') {
			continue;
		}

		if (i < str.length - 1 && str[i + 1] === '*') {
			continue;
		}

		newStr = newStr + str[i];
	}

	return newStr;
};

console.log(startOut('ab*cd'));
console.log(startOut('ab**cd'));
console.log(startOut('sm*eilly'));
